### PR TITLE
Migrate RequestGeneratorRun to prefect

### PR DESCRIPTION
**Pull request by LucasG0**

### CODE REVIEW

File: backend/infrahub/message_bus/operations/requests/generator_definition.py [hunk 1]
```diff
@@ -6,8 +6,10 @@
 from infrahub.core.constants import InfrahubKind, ValidatorConclusion, ValidatorState
```
Comment (LucasG0) [inline backend/infrahub/message_bus/operations/requests/generator_definition.py:6]: This file should be moved at a different place, I am not sure where.

File: backend/infrahub/message_bus/operations/requests/generator.py [hunk 2]
```diff
@@ -10,36 +10,38 @@
 from infrahub.core.constants import GeneratorInstanceStatus
```
Comment (LucasG0) [inline backend/infrahub/message_bus/operations/requests/generator.py:10]: This file as well as `backend/infrahub/message_bus/messages/request_generator_run.py` should be moved at a different place, I am not sure where.

Comment (dgarros) [reply backend/infrahub/message_bus/operations/requests/generator.py:10]: Let's create a directory named `generators` at the root of `backend/infrahub` for now

File: backend/infrahub/message_bus/operations/requests/generator.py [hunk 3]
```diff
@@ -65,23 +67,23 @@ async def run(message: messages.RequestGeneratorRun, service: InfrahubServices)
     await generator_instance.update(do_full_update=True)
 
 
-async def _define_instance(message: messages.RequestGeneratorRun, service: InfrahubServices) -> CoreGeneratorInstance:
-    if message.generator_instance:
+async def _define_instance(model: RequestGeneratorRun, service: InfrahubServices) -> CoreGeneratorInstance:
```
Comment (dgarros) [inline backend/infrahub/message_bus/operations/requests/generator.py:71]: would be good to add a @task decorator on this function

File: backend/infrahub/workflows/catalogue.py [hunk 4]
```diff
@@ -71,6 +71,13 @@
     tags=[WorkflowTag.DATABASE_CHANGE],
 )
 
+REQUEST_GENERATOR_RUN = WorkflowDefinition(
```
Comment (dgarros) [inline backend/infrahub/workflows/catalogue.py:74]: Need to add it to the list of workflows at the bottom of the file

**Review by dgarros — APPROVED** [review]

**Pull request merged**